### PR TITLE
tests: cover new changes with tests

### PR DESCRIPTION
**Pull request by andrii-hantkovskyi**
## Description

Adds test coverage for `migrate_from_modulestore` task functionality.

- Covers different `composition_level` values and block/container variations.
- Verifies behavior for edge cases like duplicate content, mixed child types, and empty containers.
- Ensures proper handling of `replace_existing` flag and correct target key generation.
- Confirms successful task execution paths for both components and containers.

> Test cases were created with the help of GitHub Copilot (Sonnet 4 LLM) and were thoroughly reviewed and validated by a developer.

### CODE REVIEW

File: cms/djangoapps/modulestore_migrator/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    """
+    Test cases for the modulestore migrator API.
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.library = ContentLibraryFactory()
+
+    def test_start_migration_to_library(self):
```
Comment (cmltaWt0) [inline cms/djangoapps/modulestore_migrator/tests/test_api.py:40]: Please add a case for Value error:
```python
if not (target_package_id := target_library.learning_package_id):
    raise ValueError(
         f"Cannot import {source_key} into library at {target_library_key} because the "
         "library is not connected to a learning package"
    )
```

File: cms/djangoapps/modulestore_migrator/tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        Test that the API can start a migration to a library.
+        """
+        source = ModulestoreSourceFactory()
+        user = UserFactory()
+
+        api.start_migration_to_library(
+            user=user,
+            source_key=source.key,
+            target_library_key=self.library.library_key,
+            target_collection_slug=None,
```
Comment (cmltaWt0) [inline cms/djangoapps/modulestore_migrator/tests/test_api.py:51]: Let's add a test for target_collection_slug=`some_slug`

File: cms/djangoapps/modulestore_migrator/tests/test_tasks.py [hunk 3]
```diff
@@ -0,0 +1,1174 @@
...
+@ddt.ddt
+class TestMigrateFromModulestore(ModuleStoreTestCase):
+    """
+    Test the migrate_from_modulestore task
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        lib_key = LibraryLocatorV2.from_string("lib:testorg:tes-key")
```
Comment (cmltaWt0) [inline cms/djangoapps/modulestore_migrator/tests/test_tasks.py]: Should the key be `lib:testorg:test-key` or the `lib:testorg:tes-key` is intentional?

File: cms/djangoapps/modulestore_migrator/tests/test_tasks.py [hunk 4]
```diff
@@ -0,0 +1,1174 @@
...
+            created=timezone.now(),
+            created_by=self.user.id,
+        )
+
+        children = [
+            child_version_1.publishable_entity_version,
+            child_version_2.publishable_entity_version,
+        ]
+
+        print("=" * 40)
```
Comment (cmltaWt0) [inline cms/djangoapps/modulestore_migrator/tests/test_tasks.py]: Comment just do not forget to delete this before making the PR ready for review.

File: cms/djangoapps/modulestore_migrator/tests/test_tasks.py [hunk 5]
```diff
@@ -0,0 +1,1174 @@
...
+            display_name="Test Course",
+        )
+        self.collection = Collection.objects.create(
+            learning_package=self.learning_package,
+            key="test_collection",
+            title="Test Collection",
+        )
+
+    def test_slugify_source_usage_key_course(self):
+        """Test _slugify_source_usage_key with course usage key"""
```
Comment (cmltaWt0) [inline cms/djangoapps/modulestore_migrator/tests/test_tasks.py]: Please change all docstrings according the openedx code style guide.
https://docs.openedx.org/en/latest/developers/references/developer_guide/style_guides/python-guidelines.html#docstrings

**Review by cmltaWt0 — CHANGES_REQUESTED** [review]

File: cms/djangoapps/modulestore_migrator/tests/factories.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
+"""
+Factories for creating test data for the modulestore migrator.
```
Comment (cmltaWt0) [inline cms/djangoapps/modulestore_migrator/tests/factories.py:2]: @kdmccormick What is you opinion about these factories? I believe we can try to avoid using them at all and use a real code to create all artifacts like we did in previous iteration.

File: cms/djangoapps/modulestore_migrator/tests/test_api.py (hunk 1 above)
Comment (andrii-hantkovskyi) [reply cms/djangoapps/modulestore_migrator/tests/test_api.py:40]: It's impossible to raise that Exception because of `target_library = get_library(target_library_key)`, which has `assert learning_package is not None`. I would suggest removing this check

File: cms/djangoapps/modulestore_migrator/tests/test_api.py (hunk 2 above)
Comment (andrii-hantkovskyi) [reply cms/djangoapps/modulestore_migrator/tests/test_api.py:51]: added

File: cms/djangoapps/modulestore_migrator/tests/test_tasks.py (hunk 3 above)
Comment (andrii-hantkovskyi) [reply cms/djangoapps/modulestore_migrator/tests/test_tasks.py]: should be test-key, fixed :)

File: cms/djangoapps/modulestore_migrator/tests/test_tasks.py (hunk 4 above)
Comment (andrii-hantkovskyi) [reply cms/djangoapps/modulestore_migrator/tests/test_tasks.py]: removed

File: cms/djangoapps/modulestore_migrator/tests/test_tasks.py (hunk 5 above)
Comment (andrii-hantkovskyi) [reply cms/djangoapps/modulestore_migrator/tests/test_tasks.py]: updated

File: cms/djangoapps/modulestore_migrator/tests/factories.py (hunk 6 above)
Comment (andrii-hantkovskyi) [reply cms/djangoapps/modulestore_migrator/tests/factories.py:2]: I removed ContentLibrary & LearningPackage factories (separate commit, can be reverted)

Comment (kdmccormick) [reply cms/djangoapps/modulestore_migrator/tests/factories.py:2]: > we can try to avoid using them at all and use a real code to create all artifacts like we did in previous iteration.

Yes, agreed, that would be best.

File: cms/djangoapps/modulestore_migrator/tests/test_api.py (hunk 1 above)
Comment (kdmccormick) [reply cms/djangoapps/modulestore_migrator/tests/test_api.py:40]: good point @andrii-hantkovskyi, done: https://github.com/openedx/edx-platform/pull/36873/commits/2ad36c56eed0b2a5b88daa29ebc3221b043cda74

**Review by kdmccormick — APPROVED** [review]
Thank you!
